streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -275,10 +275,6 @@ public void close(final Map<TopicPartition, Long> ackedOffsets) throws Processor
                     log.error("Failed to close state store {}: ", store.name(), e);
                 }
             }
-
-            if (ackedOffsets != null) {
-                checkpoint(ackedOffsets);
-            }
```

Comment:
If we're going to ignore the `ackedOffsets` argument here, it seems like we should revisit the interface.

There are multiple paths that lead to this method, and it's not analytically obvious why it's ok to just ignore the argument and skip checkpointing here.

If we want to move the checkpoint up in the lifecycle to `suspend`, then it seems like we should do so holistically and remove the checkpointable offsets from the paramters of `close`. What do you think?

Reply:
@vvcephei @bbejeck I checked the callers of `close(ackedOffsets)` function, 
1. Abstract task is ok since our fix is to make sure state manager could checkpoint when suspending
2. GlobalStateUpdateTask is also fine since the global state manager (using GlobalStateManagerImpl) is overriding this function who will do offset checkpoint.

So I think we should be safe to ignore this in the base class, but keep the acked offset parameter in order for subclass to checkpoint for now, which minimizing the risk of this PR. Thank you!

Reply:
What "subclasses" are you referring to? Both classes you mention are internal and not extended. I don't see any risk that we need to minimize?

Reply:
I mean GlobalStateManagerImpl implements the `close(offsets)` @mjsax

Reply:
Could wel update the code in `GlobalStateUpdateTask` to:
```
    public void close() throws IOException {
        stateMgr.checkpoint(offsets);
        stateMgr.close();
    }
```

Reply:
Oh, my point is that `stateMgr` in `GlobalStateUpdateTask` is taking in `GlobalStateManagerImpl` whose `close(offsets)` will do the checkpoint operation. So I guess we don't need to call checkpoint explicitly here right? @mjsax

Reply:
@mjsax Thoughts?

Reply:
@mjsax @abbccdda @bbejeck I left a comment below for this issue. Please take a look.